src/runtime/pyscope.cs

```diff
@@ -0,0 +1,535 @@
+            {
+                manager = PyScopeManager.Global;
+            }
+            Manager = manager;
+            obj = ptr;
+            //Refcount of the variables not increase
+            variables = Runtime.PyModule_GetDict(obj);
+            if (variables == IntPtr.Zero)
+            {
+                throw new PythonException();
+            }
+            Runtime.PyDict_SetItemString(
+                variables, "__builtins__",
+                Runtime.PyEval_GetBuiltins()
+            );
+            this.Name = this.GetVariable<string>("__name__");
+        }
+
+        public PyDict Variables()
+        {
+            Runtime.XIncref(variables);
+            return new PyDict(variables);
+        }
+
+        public PyScope NewScope()
+        {
+            var scope = Manager.Create();
+            scope.ImportAll(this);
+            return scope;
+        }
+
+        public dynamic Import(string name, string asname = null)
+        {
+            Check();
+            if (asname == null)
+            {
+                asname = name;
+            }
+            var scope = Manager.TryGet(name);
+            if(scope != null)
+            {
+                Import(scope, asname);
+                return scope;
+            }
+            PyObject module = PythonEngine.ImportModule(name);
+            Import(module, asname);
+            return module;
+        }
+
+        public void Import(PyScope scope, string asname)
+        {
+            this.SetVariable(asname, scope.obj);
+        }
+
+        /// <summary>
+        /// Import Method
+        /// </summary>
+        /// <remarks>
+        /// The import .. as .. statement in Python.
+        /// Import a module,add it to the variables dict and return the resulting module object as a PyObject.
+        /// </remarks>
+        public void Import(PyObject module, string asname = null)
+        {
+            if (asname == null)
+            {
+                asname = module.GetAttr("__name__").ToString();
+            }
+            SetVariable(asname, module);
+        }
+
+        public void ImportAll(string name)
+        {
+            var scope = Manager.TryGet(name);
+            if(scope != null)
+            {
+                ImportAll(scope);
+                return;
+            }
+            PyObject module = PythonEngine.ImportModule(name);
+            ImportAll(module);
+        }
+
+        public void ImportAll(PyScope scope)
+        {
+            int result = Runtime.PyDict_Update(variables, scope.variables);
+            if (result < 0)
+            {
+                throw new PythonException();
+            }
+        }
+
+        public void ImportAll(PyObject module)
+        {
+            if (Runtime.PyObject_Type(module.obj) != Runtime.PyModuleType)
+            {
+                throw new PyScopeException("object is not a module");
+            }
+            var module_dict = Runtime.PyModule_GetDict(module.obj);
```

Comment:
I just checked, the Runtime.PyModule_GetDict method does not increase the refcount of the module dict.